src/ray/core_worker/actor_manager.cc

```diff
@@ -91,6 +91,19 @@ std::pair<std::shared_ptr<const ActorHandle>, Status> ActorManager::GetNamedActo
       RAY_LOG(ERROR) << error_str;
       return std::make_pair(nullptr, Status::TimedOut(error_str));
     }
+  } else {
+    // When the named actor is already cached, the reference of actor_creation_return_id
+    // must be increased, so we call AddNewActorHandle again here to ensure that.
+    std::string serialized_actor_handle;
+    auto actor_handle = GetActorHandle(actor_id);
+    actor_handle->Serialize(&serialized_actor_handle);
+
+    AddNewActorHandle(
```

Comment:
We shouldn't call "New" actor handle API right? If it is cached, it means the actor handle has already been registered. Don't we need to call `AddActorHandle ` instead?

Reply:
> We shouldn't call "New" actor handle API right? If it is cached, it means the actor handle has already been registered. Don't we need to call `AddActorHandle ` instead?

Actually, the `AddNewActorHandle` is the same as `AddActorHandle` when the input argument `is_detached` is true.
But any way I think what you said is right, I will fix it. :)